lucene/core/src/test/org/apache/lucene/index/TestIndexWriter.java

```diff
@@ -4600,4 +4600,49 @@ public void testIndexWriterBlocksOnStall() throws IOException, InterruptedExcept
       }
     }
   }
+
+  public void testGetFieldNames() throws IOException {
+    Directory dir = newDirectory();
+
+    IndexWriter writer = new IndexWriter(dir, newIndexWriterConfig(new MockAnalyzer(random())));
+
+    assertEquals(Set.of(), writer.getFieldNames());
+
+    addDocWithField(writer, "f1");
+    assertEquals(Set.of("f1"), writer.getFieldNames());
+
+    // should be unmodifiable:
+    final Set<String> fieldSet = writer.getFieldNames();
+    assertThrows(UnsupportedOperationException.class, () -> fieldSet.add("cannot modify"));
+    assertThrows(UnsupportedOperationException.class, () -> fieldSet.remove("f1"));
+
+    addDocWithField(writer, "f2");
+    assertEquals(Set.of("f1", "f2"), writer.getFieldNames());
```

Comment:
Let's also assert that the original `fieldSet` has not been modified - it was a true copy and not some kind of alias over an underlying modifiable Set?

Reply:
Sounds good!
Added in: 1bc95ae7f7e

Reply:
thanks, @orenovadia !